merkletree.go

```diff
@@ -0,0 +1,274 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+var (
+	ErrInvalidTree = errors.New("[merkletree] invalid tree")
+)
+
+const (
+	EmptyBranchIdentifier = 'E'
+	LeafIdentifier        = 'L'
+)
+
+type MerkleTree struct {
+	treeNonce []byte
+	salt      []byte
+	privKey   []byte
+	pubKey    []byte
+	policies  Policies
+	root      *interiorNode
+}
+
+func InitMerkleTree(policies Policies, treeNonce, salt, pubKey, privKey []byte) *MerkleTree {
+	leftBranch := &emptyNode{
+		node: node{
+			level: 1,
+		},
+	}
+
+	rightBranch := &emptyNode{
+		node: node{
+			level: 1,
+		},
+	}
+	root := &interiorNode{
+		node: node{
+			parent: nil,
+			level:  0,
+		},
+		leftChild:  leftBranch,
+		rightChild: rightBranch,
+		leftHash:   nil,
+		rightHash:  nil,
+	}
+	leftBranch.parent = root
+	rightBranch.parent = root
+
+	m := &MerkleTree{
+		treeNonce: treeNonce,
+		salt:      salt,
+		privKey:   privKey,
+		pubKey:    pubKey,
+		policies:  policies,
+		root:      root,
+	}
+	return m
+}
+
+func (h *History) LookUp(key string) (MerkleNode, []LookUpProofNode, error) {
+	str := h.currentSTR
+	return lookUp(key, str)
+}
+
+func (h *History) LookUpInEpoch(key string, ep int64) (MerkleNode, []LookUpProofNode, error) {
+	str := h.GetSTR(ep)
+	return lookUp(key, str)
+}
+
+func lookUp(key string, str *SignedTreeRoot) (MerkleNode, []LookUpProofNode, error) {
```

Comment:
This should be the converse of `Set`,

    func (m *MerkleTree) Get(key string) MerkleNode

If you want to keep an internal `lookupNode`, similar to `insertNode`, that's fine.